### PR TITLE
Add filter structure to the Sumo Logic exporter

### PR DESCRIPTION
**Pull request by sumo-drosiek**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/sumologicexporter/filtering.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+)
+
+type filtering struct {
+	regexes []*regexp.Regexp
+}
+
+// FieldsType represents concatenated metadata
+type FieldsType string
+
+func newFiltering(fields []string) (*filtering, error) {
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/filtering.go:20]: I'd change the pointer type on which we operate on to a value (accordingly in other invocations)
It's not a fat type and there are no changes being done on struct fields.

```suggestion
func newFiltering(fields []string) (filtering, error) {
```

File: exporter/sumologicexporter/filtering.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type filtering struct {
+	regexes []*regexp.Regexp
+}
+
+// FieldsType represents concatenated metadata
+type FieldsType string
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/filtering.go:18]: To be more idiomatic I'd go for

```suggestion
// Fields represents concatenated metadata
type Fields string
```

File: exporter/sumologicexporter/filtering.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type filtering struct {
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/filtering.go:13]:
```suggestion
type filter struct {
```

🤷🏻‍♂️ ?

File: exporter/sumologicexporter/filtering_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	attributes := pdata.NewAttributeMap()
+	attributes.InsertString("key3", "value3")
+	attributes.InsertString("key1", "value1")
+	attributes.InsertString("key2", "value2")
+	attributes.InsertString("additional_key2", "value2")
+	attributes.InsertString("additional_key3", "value3")
+
+	regexes := []string{"^key[12]", "^key3"}
+	f, err := newFiltering(regexes)
+	assert.NoError(t, err)
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/filtering_test.go:22]:
```suggestion
	require.NoError(t, err)
```

No point in continuing execution

File: exporter/sumologicexporter/filtering_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+	metadata := f.GetMetadata(attributes)
+	var expected FieldsType = "key1=value1, key2=value2, key3=value3"
```
Comment (pmalek-sumo) [inline exporter/sumologicexporter/filtering_test.go:25]: This can be a `const`

File: exporter/sumologicexporter/filtering.go (hunk 1 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/filtering.go:20]: What to return in case of error? `Nil` is not a proper filtering structure. 🤔

Comment (pmalek-sumo) [reply exporter/sumologicexporter/filtering.go:20]: `filter{}`

**Review by pmalek-sumo — APPROVED** [review]

**Comment by pmalek-sumo** [comment]
Please fix lint errors

**Comment by sumo-drosiek** [comment]
bump 🤷‍♂️

**Pull request closed without merging**